tesseract_python/tesseract_python/swig/tesseract_common/types.i

```diff
@@ -0,0 +1,93 @@
+/**
+ * @file types.i
+ * @brief SWIG interface file for tesseract_common/types.h
+ *
+ * @author John Wason
+ * @date December 10, 2019
+ * @version TODO
+ * @bug No known bugs
+ *
+ * @copyright Copyright (c) 2019, Wason Technology, LLC
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+%{
+#include <tesseract_common/types.h>
+%}
+
+%define tesseract_aligned_vector(name,T)
+%template(name) std::vector<T , Eigen::aligned_allocator<T >>;
+%enddef
+
+%define tesseract_aligned_map(name,Key,Value)
+%template(name) std::map<Key, Value, std::less<Key>, Eigen::aligned_allocator<std::pair<const Key, Value>>>;
+%enddef
+
+%define tesseract_aligned_map_of_aligned_vector(name,Key,Value)
+tesseract_aligned_map(name, %arg(Key), %arg(std::vector<Value , Eigen::aligned_allocator<Value >>));
+%enddef
+
+%define tesseract_aligned_unordered_map(name,Key,Value)
+%template(name) std::unordered_map<Key,Value,std::hash<Key>,std::equal_to<Key>,Eigen::aligned_allocator<std::pair<const Key, Value>>>;
+%enddef
+
+tesseract_aligned_vector(VectorIsometry3d, Eigen::Isometry3d);
+//tesseract_aligned_vector(VectorVector3d, Eigen::Vector3d);
+//TODO: Why is this not aligned??
```

Comment:
This was pointed out to me by the FCL maintainers, that the Eigen::Vector3d is not a fixed-size vectorizable type based on the [documentation](https://eigen.tuxfamily.org/dox/group__TopicFixedSizeVectorizable.html).

Reply:
I see